### PR TITLE
feature: add key word "kspacing" and related doc and example

### LINKED ISSUE #1009: we need key word "kspacing"
## Background
KSPACING is the smallest allowed spacing between k points, it can be more convenient by setting this key word for structures  of different size.

## Describe the solution you'd like
the number of K points = max(1,|bi|/KSPACING), the unit of KSPACING is 1/Bohr

### PR DESCRIPTION
**Pull request by pxlxingliang**

### CODE REVIEW

**Comment by caic99** [comment]
Hi @pxlxingliang , would you provide a test case for it?

File: source/input.cpp [hunk 1]
```diff
@@ -541,6 +542,12 @@ bool Input::Read(const std::string &fn)
         {
             read_value(ifs, nbands_sto);
         }
+        else if (strcmp("kspacing", word) == 0)
+        {
+            read_value(ifs, kspacing);
+            if (kspacing <= 0.0)
+                ModuleBase::WARNING_QUIT("Input", "kspacing must > 0");
```
Comment (dyzheng) [inline source/input.cpp]: Don't use WARNING_QUIT in this function, it would cause the program stuck.

**Review by dyzheng — COMMENTED** [review]
Test case name should not be 190*, 101* is better

File: source/input.cpp (hunk 1 above)
Comment (pxlxingliang) [reply source/input.cpp]: ok

**Review by dyzheng — APPROVED** [review]

**Pull request merged**